robottelo/hosts.py

```diff
@@ -1753,3 +1763,169 @@ def register_host_custom_repo(self, module_org, rhel_contenthost, repo_urls):
+            )
+
+    def delete_rhsso_user(self, username):
+        """Delete the RHSSO user"""
+        user_details = self.get_rhsso_user_details(username)
+        self.execute(f"{KEY_CLOAK_CLI} delete -r {settings.rhsso.realm} users/{user_details['id']}")
+
+    def create_group(self, group_name=None):
+        """Create the RHSSO group"""
+        update_user_group = Box(RHSSO_NEW_GROUP)
+        if not group_name:
+            group_name = gen_string('alphanumeric')
+        update_user_group.name = group_name
+        self.upload_rhsso_entity(update_user_group, "create_group")
+        result = self.execute(
+            f"{KEY_CLOAK_CLI} create groups -r {settings.rhsso.realm} -f create_group"
+        )
+        return result.stdout
+
+    def delete_rhsso_group(self, group_name):
+        """Delete the RHSSO group"""
+        group_details = self.get_rhsso_groups_details(group_name)
+        self.execute(
+            f"{KEY_CLOAK_CLI} delete -r {settings.rhsso.realm} groups/{group_details['id']}"
+        )
+
+    def update_client_configuration(self, json_content, sat_obj):
+        """Update the client configuration"""
+        client_id = self.get_rhsso_client_id(sat_obj)
+        self.upload_rhsso_entity(json_content, "update_client_info")
+        update_cmd = (
+            f"{KEY_CLOAK_CLI} update clients/{client_id}"
+            "-f update_client_info -s enabled=true --merge"
+        )
+        self.execute(update_cmd)
+
+    @cached_property
+    def oidc_token_endpoint(self):
+        """getter oidc token endpoint"""
+        return (
+            f"https://{settings.rhsso.host_name}/auth/realms/"
+            f"{settings.rhsso.realm}/protocol/openid-connect/token"
+        )
+
+    def get_oidc_client_id(self, sat_obj):
+        """getter for the oidc client_id"""
+        return f"{sat_obj.hostname}-foreman-openidc"
+
+    @cached_property
+    def oidc_authorization_endpoint(self):
+        """getter for the oidc authorization endpoint"""
+        return (
+            f"https://{settings.rhsso.host_name}/auth/realms/"
+            f"{settings.rhsso.realm}/protocol/openid-connect/auth"
+        )
+
+    def get_two_factor_token_rh_sso_url(self, sat_obj):
+        """getter for the two factor token rh_sso url"""
+        return (
+            f"https://{settings.rhsso.host_name}/auth/realms/"
+            f"{settings.rhsso.realm}/protocol/openid-connect/"
+            f"auth?response_type=code&client_id={sat_obj.hostname}-foreman-openidc&"
+            "redirect_uri=urn:ietf:wg:oauth:2.0:oob&scope=openid"
+        )
+
+    def set_the_redirect_uri(self, sat_obj):
+        client_config = {
+            "redirectUris": [
+                "urn:ietf:wg:oauth:2.0:oob",
+                f"https://{sat_obj.hostname}/users/extlogin/redirect_uri",
+                f"https://{sat_obj.hostname}/users/extlogin",
+            ]
+        }
+        self.update_client_configuration(client_config, sat_obj)
+
```

Comment:
TBH I really didnt like how many helpers we have added this in one file, instead this should have been externalized to in Mixins to track them separately.

@JacobCallahan !

Reply:
I believe it was previously requested for @ColeHiggins2 to move this class into hosts.py
With that in mind, I don't think there is enough in the class definition to justify externalizing to a mixin.